### PR TITLE
PBI 27 - Discover Catalogue - Homepage

### PR DESCRIPTION
**Pull request by matthewhjt**
CHANGES:
- Add findTrendingItineraries
Trending itineraries are based on the number of likes on the itinerary, limited to only get 10 itineraries
image

- 100% Coverage
image

Note: 
- To test, we must first populate the ItineraryLike first

## Summary by CodeRabbit

- **New Features**
  - Introduced a new endpoint to fetch trending itineraries, displaying the top 10 published itineraries ordered by number of likes. Each itinerary now shows a likes count and user profile details.
- **Tests**
  - Added comprehensive tests for the trending itineraries feature in both service and controller layers, covering successful retrieval, empty results, error handling, and response structure.

### CODE REVIEW

File: src/itinerary/itinerary.controller.spec.ts [hunk 1]
```diff
@@ -1992,4 +1993,156 @@ describe('ItineraryController', () => {
...
+      mockItineraryService.findTrendingItineraries.mockResolvedValue(
+        mockTrendingItineraries
+      )
+      mockResponseUtil.response.mockReturnValue(mockResponse)
+
+      // Act
+      const result = await controller.findTrendingItineraries()
+
+      // Assert
+      expect(result.data.itineraries[0]).toHaveProperty('id')
```
Comment (mmalkindi) [inline src/itinerary/itinerary.controller.spec.ts:2139]: assertion bisa di refactor supaya lebih mudah untuk di maintain

**Review by mmalkindi — COMMENTED** [review]
Good progress, but I have some (non fatal) issues with your code

File: src/itinerary/itinerary.controller.spec.ts (hunk 1 above)
Comment (matthewhjt) [reply src/itinerary/itinerary.controller.spec.ts:2668]: Halo, terima kasih atas masukkanya. Saya sudah refactor kodenya. Sepertinya test tersebut redundant dan tidak berpengaruh ke coverage maupun edge case tertentu. Terima kasih!
